dotnet/src/Azure.Iot.Operations.Protocol/Telemetry/IncomingTelemetryMetadata.cs

```diff
@@ -109,60 +111,5 @@ internal IncomingTelemetryMetadata(MqttApplicationMessage message, uint packetId
 
             PacketId = packetId;
         }
-
-        public CloudEvent GetCloudEvent()
-        {
-            string safeGetUserProperty(string name)
-            {
-                return UserData.FirstOrDefault(
-                                p => p.Key.Equals(name.ToLowerInvariant(),
-                                StringComparison.OrdinalIgnoreCase)).Value ?? string.Empty;
-            }
-
-            string specVersion = safeGetUserProperty(nameof(CloudEvent.SpecVersion).ToLowerInvariant());
-
-
-            if (!specVersion.Equals("1.0", StringComparison.Ordinal))
-            {
-                throw new ArgumentException($"Could not parse cloud event from telemetry: Only version 1.0 supported. Version provided: {specVersion}");
```

Comment:
is there any way for the customer to get this detail on why the cloud event wasn't valid anymore? And is all of the validation that was present here still present on the generic one?